### PR TITLE
Rerun 0.18.1

### PR DESCRIPTION
**Pull request by teh-cmc**
- [x]  Revert breaking changes (including BGR support)
- [x] `cargo semver-checks` passes
- [x] Update changelog
- [x] https://github.com/rerun-io/rerun/issues/7294 has been tested
- [x] Ready to release

[Rendered changelog](https://github.com/rerun-io/rerun/blob/release-0.18.1/CHANGELOG.md#0181---bug-fixes-and-performance-improvements)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,8 +1,59 @@
...
+- Fix python version updating script [#7213](https://github.com/rerun-io/rerun/pull/7213)
+- Fix url to event.rs in `rerun analytics details` [#7230](https://github.com/rerun-io/rerun/pull/7230)
+- BGR(A) image format support [#7238](https://github.com/rerun-io/rerun/pull/7238)
+- Fix header levels in `sdk-operating-modes.md` [#7231](https://github.com/rerun-io/rerun/pull/7231)
+- Hide time controls if there is only one time point on a timeline [#7241](https://github.com/rerun-io/rerun/pull/7241)
+- Speed up data density graph by rendering them more coarsly [#7229](https://github.com/rerun-io/rerun/pull/7229)
+- Remove `Datatype` and `DatatypeBatch` [#7256](https://github.com/rerun-io/rerun/pull/7256)
+- Fix typo in spaces-and-transforms.md [#7255](https://github.com/rerun-io/rerun/pull/7255)
+- Remove `Loggable{Batch}::arrow_field` [#7257](https://github.com/rerun-io/rerun/pull/7257)
+- Remove `LoggableBatch::num_instances` [#7258](https://github.com/rerun-io/rerun/pull/7258)
```
Comment (emilk) [inline CHANGELOG.md:40]: These changes won't be semver compatible (hopefully `cargo semver-checks` should catch this)

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,8 +1,59 @@
...
+- re_arrow2 0.17.5 [#7262](https://github.com/rerun-io/rerun/pull/7262)
```
Comment (emilk) [inline CHANGELOG.md:41]: This PR should have called out the user-facing change, i.e. the memory capacity leak

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,8 +1,59 @@
...
+- Fix main (`cargo insta review`) [#7264](https://github.com/rerun-io/rerun/pull/7264)
```
Comment (emilk) [inline CHANGELOG.md:42]: Wait, this is marked "exclude form changelog", so how did it make it into the changelog? Is our changelog script broken?

**Review by emilk — COMMENTED** [review]
It seems like our changelog script is not heeding the "exclude from changelog" label 🤔

**Review by emilk — COMMENTED** [review]
It seems like our changelog script is not heeding the "exclude from changelog" label 🤔

**Comment by teh-cmc** [comment]
> It seems like our changelog script is not heeding the "exclude from changelog" label 🤔

Ha? Maybe I messed up, I've been pushing the multitasking a bit too far on this one...

File: CHANGELOG.md (hunk 1 above)
Comment (teh-cmc) [reply CHANGELOG.md:40]: I still think we should ship these anyway -- we didn't even use them ourselves.

I can highlight them in the changelog.

**Comment by teh-cmc** [comment]
Okay so apparently my Github token expired, and the script returns a weird changelog when that happens oO

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,8 +1,66 @@
...
+
+Bug fixes and performance improvements for Rerun 0.18.0.
+
+### 🔎 Details
+
+### ⚠️ Breaking changes
+- Remove unused `Datatype` and `DatatypeBatch` [#7256](https://github.com/rerun-io/rerun/pull/7256)
+- Remove unused `Loggable{Batch}::arrow_field` [#7257](https://github.com/rerun-io/rerun/pull/7257)
+- Remove unused `LoggableBatch::num_instances` [#7258](https://github.com/rerun-io/rerun/pull/7258)
+- Remove unused `Loggable::extended_arrow_datatype` [#7260](https://github.com/rerun-io/rerun/pull/7260)
```
Comment (emilk) [inline CHANGELOG.md:17]: ⚠️ We shouldn't ship breaking changes in a patch release! We can just `git revert` these instead

File: CHANGELOG.md (hunk 1 above)
Comment (emilk) [reply CHANGELOG.md]: Better to get into the habit of not shipping breaking changes in patch releases imho

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,8 +1,66 @@
...
+- Update `time` crate to 0.3.36, fixing compilation on newer Rust versions [#7228](https://github.com/rerun-io/rerun/pull/7228)
+- re_arrow2 0.17.5 (fixes memory leak) [#7262](https://github.com/rerun-io/rerun/pull/7262)
+
+#### 🤷‍ Other
+- Fix typo in spaces-and-transforms.md [#7255](https://github.com/rerun-io/rerun/pull/7255) (thanks [@NickLaurenson-Visionick](https://github.com/NickLaurenson-Visionick)!)
+- Rerun 0.18.1 [2d87f50aafa63c0a89cbed0f683e4aed1330052f](https://github.com/rerun-io/rerun/commit/2d87f50aafa63c0a89cbed0f683e4aed1330052f)
+- changelog [35a5e2f0ddb975760f3b508ca00b5e15c30372b0](https://github.com/rerun-io/rerun/commit/35a5e2f0ddb975760f3b508ca00b5e15c30372b0)
+- have to fix the pixi.lock for some reason? im not sure [80af97e61460c91ab312751f052ca2e33b0d9947](https://github.com/rerun-io/rerun/commit/80af97e61460c91ab312751f052ca2e33b0d9947)
+- typo [4711939c6f77a7802d07fb4cf97c40c11cc39fca](https://github.com/rerun-io/rerun/commit/4711939c6f77a7802d07fb4cf97c40c11cc39fca)
+
```
Comment (emilk) [inline CHANGELOG.md:61]:
```suggestion
```

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,8 +1,66 @@
...
+
+## [0.18.0](https://github.com/rerun-io/rerun/compare/0.17.0...HEAD) - Ingestion speed and memory footprint
```
Comment (emilk) [inline CHANGELOG.md:63]:
```suggestion
## [0.18.0](https://github.com/rerun-io/rerun/compare/0.17.0...0.18.0) - Ingestion speed and memory footprint
```

File: CHANGELOG.md [hunk 7]
```diff
@@ -1,8 +1,66 @@
 # Rerun changelog
 
+
 ## [Unreleased](https://github.com/rerun-io/rerun/compare/latest...HEAD)
 
-## [0.18.0](https://github.com/rerun-io/rerun/compare/0.16.1...HEAD) - Ingestion speed and memory footprint
+
+## [0.18.1](https://github.com/rerun-io/rerun/compare/0.17.0...HEAD) - Bug fix
```
Comment (emilk) [inline CHANGELOG.md:7]:
```suggestion
## [0.18.1](https://github.com/rerun-io/rerun/compare/0.18.0...0.18.1) - Bug fix
```

File: CHANGELOG.md [hunk 8]
```diff
@@ -1,8 +1,59 @@
...
+- Convert most remaining `DataUI` component implementations to component uis in `re_component_ui` [#7247](https://github.com/rerun-io/rerun/pull/7247)
+- Fix references to closed tickets [#7219](https://github.com/rerun-io/rerun/pull/7219)
+- Fix gamma (srgb EOTF) for GLTF via `Asset3D` embedded rgb(a) textures [#7251](https://github.com/rerun-io/rerun/pull/7251)
+- Consistently call it "latest-at", with a dash [#7240](https://github.com/rerun-io/rerun/pull/7240)
+- Link to the per-language docs for `send_columns` [#7244](https://github.com/rerun-io/rerun/pull/7244)
+- Remove references to justfile [#7243](https://github.com/rerun-io/rerun/pull/7243)
+- Show single-line editor if there is no multiline editor [#7211](https://github.com/rerun-io/rerun/pull/7211)
+- Fix python version updating script [#7213](https://github.com/rerun-io/rerun/pull/7213)
+- Fix url to event.rs in `rerun analytics details` [#7230](https://github.com/rerun-io/rerun/pull/7230)
+- BGR(A) image format support [#7238](https://github.com/rerun-io/rerun/pull/7238)
```
[the remark this answers is not in the document]
Comment (emilk) [reply CHANGELOG.md:37]: And `cargo semver-checks` catches it ❤️ 

```
> cargo +1.77.0 semver-checks -p re_types
…
--- failure enum_variant_added: enum variant added on exhaustive enum ---

Description:
A publicly-visible enum without #[non_exhaustive] has a new variant.
        ref: https://doc.rust-lang.org/cargo/reference/semver.html#enum-variant-new
       impl: https://github.com/obi1kenobi/cargo-semver-checks/tree/v0.34.0/src/lints/enum_variant_added.ron

Failed in:
  variant ColorModel:BGR in /Users/emilk/code/rerun/rerun/crates/store/re_types/src/datatypes/../datatypes/color_model.rs:41
  variant ColorModel:BGRA in /Users/emilk/code/rerun/rerun/crates/store/re_types/src/datatypes/../datatypes/color_model.rs:45
```

**Comment by emilk** [comment]
PRs have been reverted, changelog updated, and `re_types`, `re_sdk` and `rerun` all pass `cargo semver-checks` now :white_check_mark:

**Comment by teh-cmc** [comment]
Cherry picking the changelog back to main then closing this.

**Pull request closed without merging**